testsuite/ui/views/master/audience/tenant.py

```diff
@@ -0,0 +1,137 @@
+"""View representations of Tenants pages"""
+from widgetastic.widget import TextInput, GenericLocatorWidget, Text
+
+from testsuite.ui.navigation import step
+from testsuite.ui.views.master.audience import BaseMasterAudienceView
+from testsuite.ui.widgets import AudienceTable
+from testsuite.ui.widgets.buttons import ThreescaleDeleteButton, \
+    ThreescaleEditButton, ThreescaleSubmitButton, ThreescaleSearchButton
+
+
+class TenantView(BaseMasterAudienceView):
+    """View representation of Tenants Listing page"""
+    # TODO search will be separated into the AudienceTable Widget later.
+    path_pattern = '/buyers/accounts'
+    new_account = Text("//a[@href='/p/admin/accounts/new']")
+    table = AudienceTable("//*[@id='buyer_accounts']")
+    search_button = ThreescaleSearchButton()
+    search_bar = TextInput(id="search_query")
+
+    def search(self, value: str):
+        """Search in Tenant table by given value"""
+        self.search_bar.fill(value)
+        self.search_button.click()
+
+    @step("TenantNewView")
+    def new(self):
+        """Create new Tenant"""
+        self.new_account.click()
+
+    @step("TenantDetailView")
+    def detail(self, account):
+        """Opens detail Account by ID"""
+        self.table.row(_row__attr=('id', f'account_{account.entity_id}')).grouporg.click()
+
+    def prerequisite(self):
+        return BaseMasterAudienceView
+
+    @property
+    def is_displayed(self):
+        return BaseMasterAudienceView.is_displayed.fget(self) and self.new_account.is_displayed and \
+               self.table.is_displayed and self.path_pattern in self.browser.url
+
+
+class TenantDetailView(BaseMasterAudienceView):
+    """View representation of Tenant detail page"""
+    path_pattern = '/buyers/accounts/{account_id}'
+    edit_button = ThreescaleEditButton()
+    applications_button = Text("//*[contains(@title,'applications')]")
+    public_domain = Text('//*[@id="twoCol"]/div[1]/div/table/tbody/tr[2]/td/a')
+    admin_domain = Text('//*[@id="twoCol"]/div[1]/div/table/tbody/tr[3]/td/a')
+    suspend_button = Text('//*[@id="twoCol"]/div[1]/div/table/tbody/tr[6]/td/a')
+
+    def __init__(self, parent, account):
+        super().__init__(parent, account_id=account.entity_id)
+
+    @step("TenantEditView")
+    def edit(self):
+        """Edit account"""
+        self.edit_button.click()
+
+    def suspend_or_resume(self):
```

Comment:
We need a method that would just suspend or just resume the tenant. This is not deterministic.